server/service/testing_client.go

```diff
@@ -342,7 +342,7 @@ func (ts *withServer) LoginSSOUser(username, password string) (fleet.Auth, strin
 }
 
 func (ts *withServer) LoginMDMSSOUser(username, password string) *http.Response {
-	_, res := ts.loginSSOUser(username, password, "/api/v1/fleet/mdm/sso", http.StatusTemporaryRedirect)
+	_, res := ts.loginSSOUser(username, password, "/api/v1/fleet/mdm/sso", http.StatusSeeOther)
```

Comment:
As a follow up PR (not blocking this merge), I think it would be good to add an integration test that catches non-200 errors in the SSO flow from the perspective of the embedded frontend client.

Reply:
we have some (look for when we extract `errors` from the redirect URI query) but we could add more for sure 💯 

I think we didn't catch this (and we won't catch similar errors) because when we setup the test server, we only add a handler for the API endpoints:

https://github.com/fleetdm/fleet/blob/d213f843e78f54ee8e7338f1018c45228bd68590/server/service/testing_utils.go#L347-L348

Reply:
That makes sense. I figured we probably didn't have the test env configured for this flow. I wonder how complicated it would be to add something that would serve the route for automated testing. Maybe it just serves some dummy html rather than the full frontend app but allow us to test things like the response codes. It would be nice to have some coverage for the flow that wasn't reliant on the outsourced e2e.